hazelcast-jet-core/src/main/java/com/hazelcast/jet/core/test/TestOutbox.java

```diff
@@ -131,6 +158,57 @@ public boolean offerToSnapshot(Object key, Object value) {
         return snapshotQueue;
     }
 
+    /**
+     * Move all items from the queue to the {@code target} collection and make
+     * the outbox available to accept more items. Also calls {@link
+     * #resetBatch()}. If you have limited capacity outbox, you need to call
+     * this regularly.
+     *
+     * @param queueOrdinal the queue from Outbox to drain
+     * @param target target list
+     * @param logItems whether to log drained items to {@code System.out}
+     */
+    public <T> void drainQueueAndReset(int queueOrdinal, Collection<T> target, boolean logItems) {
+        drainInternal(queue(queueOrdinal), target, logItems);
+    }
+
+    /**
+     * Move all items from the snapshot queue to the {@code target} collection
+     * and make the outbox available to accept more items. Also calls {@link
+     * #resetBatch()}. If you have limited capacity outbox, you need to call
+     * this regularly.
+     *
+     * @param target target list
+     * @param logItems whether to log drained items to {@code System.out}
+     */
+    public <T> void drainSnapshotQueueAndReset(Collection<T> target, boolean logItems) {
+        drainInternal(snapshotQueue(), target, logItems);
+    }
+
+    private <T> void drainInternal(Queue<?> q, Collection<T> target, boolean logItems) {
+        for (Object o; (o = q.poll()) != null; ) {
+            target.add((T) o);
+            if (logItems) {
+                System.out.println(LocalTime.now() + " Output: " + o);
+            }
+        }
+        resetBatch();
+    }
+
+    /**
+     * Call this method after any of the {@code offer()} methods returned
+     * {@code false} to be able to offer again. Method is called automatically
+     * from {@link #drainQueueAndReset} and {@link #drainSnapshotQueueAndReset}
+     * methods.
+     */
+    public void resetBatch() {
```

Comment:
shouldn't this also be called `reset` now?

Reply:
fixed